etcdserver/raft.go

```diff
@@ -159,10 +159,16 @@ func (r *raftNode) start(s *EtcdServer) {
 						if r.s.stats != nil {
 							r.s.stats.BecomeLeader()
 						}
+						if r.s.compactor != nil {
+							r.s.compactor.Resume()
+						}
 					} else {
 						if r.s.lessor != nil {
 							r.s.lessor.Demote()
 						}
+						if r.s.compactor != nil {
+							r.s.compactor.Pause()
```

Comment:
we want the leader to control the compaction. the follower could help though. (for example, the leader might be a restarted node and just reset the timer and fail to compact in time.) but i feel this is somewhat overly complicated the issue. i want to see how the simplest solution goes before making this more advanced.